File: frontend/src/types/equipment.ts
export const EquipmentCategory = {
  MICROCONTROLLERS: 'MICROCONTROLLERS',
  VR: 'VR',
  NETWORKING: 'NETWORKING',
  SENSORS: 'SENSORS',
  ROBOTICS: 'ROBOTICS',
  COMPUTING: 'COMPUTING',
} as const
export type EquipmentCategory = (typeof EquipmentCategory)[keyof typeof EquipmentCategory]

export const EquipmentStatus = {
  AVAILABLE: 'AVAILABLE',
  RESERVED: 'RESERVED',
  MAINTENANCE: 'MAINTENANCE',
} as const
export type EquipmentStatus = (typeof EquipmentStatus)[keyof typeof EquipmentStatus]

export interface Equipment {
  id: number
  name: string
  serialNumber: string
  macAddress: string | null
  category: EquipmentCategory
  status: EquipmentStatus
  createdAt: string
  updatedAt: string
}

export interface PageResponse<T> {
  content: T[]
  totalPages: number
  totalElements: number
  last: boolean
  first: boolean
  size: number
  number: number
  numberOfElements: number
  empty: boolean
}

File: src/main/java/equipment_api/entity/Equipment.java
package equipment_api.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDateTime;

@Entity
@Table(name = "equipment")
public class Equipment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "El nombre del equipo es obligatorio")
    @Size(max = 120, message = "El nombre no puede superar los 120 caracteres")
    @Column(nullable = false)
    private String name;

    /**
     * Identificador fisico del equipo. El enunciado admite "numero de serie o
     * MAC": este campo acepta ambos formatos y es unico en la tabla.
     */
    @NotBlank(message = "El numero de serie (o MAC) es obligatorio")
    @Size(max = 80, message = "El numero de serie no puede superar los 80 caracteres")
    @Column(nullable = false, unique = true)
    private String serialNumber;

    @NotNull(message = "La categoria es obligatoria (MICROCONTROLLERS, VR o NETWORKS)")
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EquipmentCategory category;

    @NotNull(message = "El estado es obligatorio (AVAILABLE, RESERVED o MAINTENANCE)")
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private EquipmentStatus status;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    public Equipment() {
    }

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    // Getters y Setters

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public void setSerialNumber(String serialNumber) {
        this.serialNumber = serialNumber;
    }

    public EquipmentCategory getCategory() {
        return category;
    }

    public void setCategory(EquipmentCategory category) {
        this.category = category;
    }

    public EquipmentStatus getStatus() {
        return status;
    }

    public void setStatus(EquipmentStatus status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }
}